querybuilder/rule_group_test.go

```diff
@@ -66,7 +68,10 @@ func TestRuleGroupEvaluate(t *testing.T) {
 
 			rg := RuleGroup{Condition: rules["condition"], Rules: rules["rules"]}
 
-			if got := rg.Evaluate(parseJson(input.dataset)); got != input.want {
+			if got, err := rg.Evaluate(parseJson(input.dataset)); got != input.want {
+				if err != nil {
+					errors.Errorf("Caught Error %s", err)
```

Comment:
does this fail this test, when there is an error? Verify

Reply:
If there's an error and `got` matches with `input.want`, we aren't checking if err is there or not. If there's an error, it still goes into the block and executes the statements inside. That's why I did 2 checks - error check and if result and expected result match. 

If I club both conditions check into a single `if`, then what should be the error message? The error message returned by `err` or that result doesn't match.